envi/archs/arm/disasm.py

```diff
@@ -218,7 +218,7 @@ def p_dp_imm_shift(opval, va):
 qop_mnem = ('qadd','qsub','qdadd','qdsub')
 smla_mnem = ('smlabb','smlabt','smlatb','smlatt',)
 smlal_mnem = ('smlalbb','smlalbt','smlaltb','smlaltt',)
-smul_mnem = ('smulbb','smulbt','smultb','smultt',)
+smul_mnem = ('smulbb','smultb','smulbt','smultt',)  #bt and tb are backwards. Note xy bits are actually yx
```

Comment:
Wasn't sure if the swap should have been done here or at the decoding section but seemed like it was easier to do here and fix later if the emulation was backwards.

Date: Fri, 3 Jun 2016 05:52:32 -0700
From: notifications@github.com
To: vivisect@noreply.github.com
CC: redheadedrod@hotmail.com; author@noreply.github.com
Subject: Re: [atlas0fd00m/vivisect] Update unit tests for opcodes (#3)

In envi/archs/arm/disasm.py:

> @@ -218,7 +218,7 @@ def p_dp_imm_shift(opval, va):
>  qop_mnem = ('qadd','qsub','qdadd','qdsub')
>  smla_mnem = ('smlabb','smlabt','smlatb','smlatt',)
>  smlal_mnem = ('smlalbb','smlalbt','smlaltb','smlaltt',)
> -smul_mnem = ('smulbb','smulbt','smultb','smultt',)
> +smul_mnem = ('smulbb','smultb','smulbt','smultt',)  #bt and tb are backwards. Note xy bits are actually yx

another good catch.  since this encoding is an oddball, please go to the parser and change "xy" to "yx" so it's easy to see the difference.


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.